src/components/LinkShortener.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { pallete } from "../misc/pallete";
+import Button from "./Button";
+import PropTypes from "prop-types";
+import Loader from "./Loader";
+const linkRegex =
+  /[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/;
+
+export default function LinkShortener({ setLinkArr, scrollToLink }) {
+  const [nextLink, setNextLink] = useState("");
+  const [isError, setIsError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleChange = (e) => {
+    setNextLink(e.target.value);
+  };
+  const shortenLink = async () => {
```

Comment:
funkcja powinna odpowiadać za jedną rzecz - tutaj mamy zarówno sprawdzenie, czy podany link jest na pewno linkiem, później mamy zapytanie do API i później jeszcze przetworzenie uzyskanych z API danych.
Może być jedna funkcja, która to skleja w całość, ale pojedyncze fragmenty powinny być porozdzielane do ososbnych funkcji